### ISSUE TITLE
MIMIC III download is way too slow

### ISSUE THREAD

**Issue by asmgx** [MIT-LCP/mimic-code#600]
### Prerequisites

* [ X] Put an X between the brackets on this line if you have done all of the following:
    * Checked the online documentation: https://mimic.physionet.org/about/mimic/
    * Checked that your issue isn't already addressed: https://github.com/MIT-LCP/mimic-code/issues?utf8=%E2%9C%93&q=

### Description

MIMIC III download is way too slow.
The file is 6GB+ but download speed is 5KB/sec

How can I fix that?

**Comment by arasekh** [comment]
Yes, I have also the same problem. When I downloaded the MIMIC-CXR dataset using Wget on our Ubuntu server, it downloaded only 60 MB in 6 mins!! By the way speed test shows a much higher bandwidth for the server.

I tested the MIMIC-IV dataset on my laptop with LAN(high bandwidth). When I downloaded the zip file with my browser it downloaded very fast, but when I used wget on my terminal again was suuuper slow!!.

I searched and guessed that wget is slow with Https. But maybe downloading a huge number of small files is the issue. 
Anyway did you have any suggestions? It might take 60 days to download the MIMIC-CXR with this speed!

**Comment by arasekh** [comment]
By the way I tested these commands:
And the same original command in this link, but the same problem:
wget -r -N -c -np --user arasekh --ask-password https://physionet.org/files/mimic-cxr/2.0.0/


And this one based on the suggested solutions on the web:
wget -r -N -c -np --user arasekh --http-password=*** --inet4-only -v --no-check-certificate --tries=10 https://physionet.org/files/mimic-cxr-jpg/2.0.0/

**Comment by arasekh** [comment]
> ### Prerequisites
> * [ X] Put an X between the brackets on this line if you have done all of the following:
>   
>   * Checked the online documentation: https://mimic.physionet.org/about/mimic/
>   * Checked that your issue isn't already addressed: https://github.com/MIT-LCP/mimic-code/issues?utf8=%E2%9C%93&q=
> 
> ### Description
> MIMIC III download is way too slow. The file is 6GB+ but download speed is 5KB/sec
> 
> How can I fix that?

Did you find any solution?

**Comment by bemoody** [comment]
Please do not use options `--inet4-only` and `--no-check-certificate`.  Whoever suggested that has no idea what they're talking about.  These options have nothing to do with speed, I promise.

**Comment by alistairewj** [comment]
I suspect what's happening is that wget is downloading all the index files first, and there are many hundreds of thousands of them. As a result, the initial bandwidth is slow (60 mb/6 minutes). However after it completes the index files, it begins to download the actual data files. This has much less overhead per file as the files are larger, and you should see speeds be more reasonable.

Unfortunately there isn't really a way to tell `wget` to ignore index files. You can ask it to remove index files after the command finishes, but that's about it.

**Comment by TheCedarPrince** [comment]
Oddly, I can confirm this as well @alistairewj and @AliRasekh -- I tested the server to make sure I could use an analogous `wget` command for another file (1GB) and there I was able to download the 1GB file within seconds whereas downloading the CHARTEVENTS file for MIMIC III (4.6GB) is taking hours. I tested another machine to see if this was just a machine issue and saw the same performance issues. 

I am not sure if the index files is what is causing the issue. I could download the ADMISSIONS, CALLOUT, and CAREGIVER files but they showed the exact same slowness as the other files. This feels like some sort of strange bandwidth or access degradation taking place. Anything else I could help or report on about this?

**Comment by TheCedarPrince** [comment]
Ah thanks @tompollard -- this is a great workaround and thanks for the speedy response! I so appreciate the work that you and everyone else do on making this awesome resource available to the world. Have a great day!

**Comment by ChennaKeshavaReddy4644** [comment]
> By the way I tested these commands: And the same original command in this link, but the same problem: wget -r -N -c -np --user arasekh --ask-password https://physionet.org/files/mimic-cxr/2.0.0/
> 
> And this one based on the suggested solutions on the web: wget -r -N -c -np --user arasekh --http-password=*** --inet4-only -v --no-check-certificate --tries=10 https://physionet.org/files/mimic-cxr-jpg/2.0.0/

bro how many days it took to download the mimic cxr database

**Comment by abdulfatir** [comment]
Is there a reason why one there's no option to `wget` the zip file?

**Comment by ronymacwan** [comment]
I am getting Access Denied error in the bucket. Is anyone facing the same issue?

**Comment by briangow** [comment]
@ronymacwan , it doesn't sound like you are dealing with a slow download, so it'd be best to open a new issue (or if you think the problem is specific to your setup to email contact@physionet.org instead). We would also likely need more detail about your issue - are you trying to download from AWS? What is the exact error message?
